### PR TITLE
[Lock] Create flock directory

### LINKED ISSUE #35697: [Lock] FlockStore does not create the directory
**Symfony version(s) affected**: all

**Description**  
The `FlockStore` does not create the directory if it does not exist.
I don't know if it is made on purpose...

When declaring the lock system with framework bundle:
```yaml
# /packages/framework.yaml
framework:
    lock:
        file: 'flock://%kernel.project_dir%/var/lock'
```
This results in `The directory "/path/to/var/lock" is not writable.` althought the `./var` directory is writable.
Why don't we check if the directory exists, and create it if it doesn't?

**How to reproduce**  
Declare the lock with the given yaml, when directory `./var/lock` does not exist.

**Possible Solution**  
Create the directory if it doesn't exist, in the `FlockStore` constructor

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #35697
| License       | MIT
| Doc PR        | -

Automatically create the flock directory when does not exists

### CODE REVIEW

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jderusse.

File: src/Symfony/Component/Lock/Tests/Store/FlockStoreTest.php [hunk 1]
```diff
@@ -46,14 +46,23 @@ public function testConstructWhenRepositoryDoesNotExist()
...
         new FlockStore('/');
     }
 
+    public function testConstructWithSubdir()
+    {
+        if (!getenv('USER') || 'root' === getenv('USER')) {
+            $this->markTestSkipped('This test will fail if run under superuser');
+        }
+
+        new FlockStore(sys_get_temp_dir().'/sf-flock');
```
Comment (stof) [inline src/Symfony/Component/Lock/Tests/Store/FlockStoreTest.php:63]: this test has no assertion, which marks it as risky

File: src/Symfony/Component/Lock/Tests/Store/FlockStoreTest.php [hunk 2]
```diff
@@ -46,14 +46,23 @@ public function testConstructWhenRepositoryDoesNotExist()
...
             $this->markTestSkipped('This test will fail if run under superuser');
         }
 
         new FlockStore('/');
     }
 
+    public function testConstructWithSubdir()
+    {
+        if (!getenv('USER') || 'root' === getenv('USER')) {
+            $this->markTestSkipped('This test will fail if run under superuser');
```
Comment (stof) [inline src/Symfony/Component/Lock/Tests/Store/FlockStoreTest.php:60]: is this actually true ? It does not rely on some directories not being writable.

File: src/Symfony/Component/Lock/Tests/Store/FlockStoreTest.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/Lock/Tests/Store/FlockStoreTest.php:63]: It's worth than that, it makes our CI fails currently :)

Comment (jderusse) [reply src/Symfony/Component/Lock/Tests/Store/FlockStoreTest.php:63]: i'm on it

File: src/Symfony/Component/Lock/Tests/Store/FlockStoreTest.php (hunk 2 above)
Comment (jderusse) [reply src/Symfony/Component/Lock/Tests/Store/FlockStoreTest.php:60]: good catch. fixed by https://github.com/symfony/symfony/pull/39721

**Pull request merged**